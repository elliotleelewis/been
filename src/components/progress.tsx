import { type FC, memo } from 'react';

interface Props {
	complete: number;
}

export const Progress: FC<Props> = memo(({ complete }) => (
	<div className="relative size-5">
		<svg
			className="-rotate-90 size-full"
			viewBox="0 0 36 36"
			xmlns="http://www.w3.org/2000/svg"
		>
			<title>{Math.round(complete * 1000) / 10}% complete.</title>
			<circle
				cx="18"
				cy="18"
				r="16"
				fill="none"
				className="stroke-current text-gray-500 dark:text-neutral-500"
				strokeWidth="4"
			/>
			<circle
				cx="18"
				cy="18"
				r="16"
				fill="none"
				className="stroke-current text-primary dark:text-primary"
				strokeWidth="4"
				strokeDasharray="100"
				strokeDashoffset={(1 - complete) * 100}
				strokeLinecap="round"
			/>
		</svg>
	</div>
));
Progress.displayName = 'Progress';
