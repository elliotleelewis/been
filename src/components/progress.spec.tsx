import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { Progress } from './progress';

describe('Progress', () => {
	it('should render', () => {
		const result = render(<Progress complete={0} />);

		expect(result.asFragment()).toMatchSnapshot();
	});
});
