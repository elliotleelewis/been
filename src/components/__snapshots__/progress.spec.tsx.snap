// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress > should render 1`] = `
<DocumentFragment>
  <div
    class="relative size-5"
  >
    <svg
      class="size-full -rotate-90"
      viewBox="0 0 36 36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        0% complete.
      </title>
      <circle
        class="stroke-current text-gray-200 dark:text-neutral-500"
        cx="18"
        cy="18"
        fill="none"
        r="16"
        stroke-width="4"
      />
      <circle
        class="stroke-current text-primary dark:text-primary"
        cx="18"
        cy="18"
        fill="none"
        r="16"
        stroke-dasharray="100"
        stroke-dashoffset="100"
        stroke-linecap="round"
        stroke-width="4"
      />
    </svg>
  </div>
</DocumentFragment>
`;
